ui/apps/platform/src/hooks/useRestMutation.ts

```diff
@@ -0,0 +1,107 @@
+import { useState } from 'react';
+
+export type RequestStatus = 'idle' | 'loading' | 'success' | 'error';
+
+export type MutationOptions<Result> = {
+    /** A function to call when the mutation is successful */
+    onSuccess?: (data: Result) => void;
+    /** A function to call when the mutation fails */
+    onError?: (error: unknown) => void;
+    /** A function to call after the mutation is successful or fails */
+    onSettled?: (data: Result | undefined, error: unknown) => void;
```

Comment:
The main benefit of having this is to avoid duplicate code in both the `onSuccess` and `onError` callbacks to cover both cases. The benefit is probably minimal, but saves a little bit of code at the call site and can potentially help avoid lost logic during refactoring.

From the other point of view, having the two callbacks instead of using only `onSettled` helps quite a bit with readability. It is easier to see at a glance if we are handling the good/bad code paths, and it avoids having to check if the values are defined for the two arguments to `onSettled`. (It avoid ambiguity too, since both of the arguments to `onSettled` could be `undefined` regardless of the success state.)